app/lib/summary_stats_utils.rb

```diff
@@ -88,4 +92,104 @@ def self.ingest_run_count(start_date: Time.zone.today, end_date: Time.zone.today
     end
     ingest_jobs
   end
+
+  # returns an array of hashes, each with title, accession, study_owner,
+  # one entry in the array for each study that has been deleted during the time frame
+  def self.deleted_studies_info(start_date: Time.zone.today, end_date: Time.zone.today + 1.day)
+    deletions = HistoryTracker.trackers_by_date(Study, action: 'destroy', start_time: start_date, end_time: end_date)
+    deletion_info = deletions.map do |tracker|
+      {
```

Comment:
There are several places where we're assembling this info hash - it should be extracted into a helper method that can also take extra args (like `other_studies`, `update_info`, etc).  Passing them with a double splat (`**`) would allow you to dynamically merge in as many extra hash elements like so:

```
def format_study_info(tracker, **extra_info)
  {
    title: tracker.original['name'],
    accession: tracker.original['accession'],
    study_owner: User.find_by(id: tracker.original['user_id']).try(:email)
  }.merge(extra_info)
end

SummaryStatsUtils.format_study_info(study, foo: 'bar', bing: 'baz')
=> {:name=>"My Study", :accession=>"SCP1234", :study_owner=> 'fake@email.com', :foo=>"bar", :bing=>"baz"}
```

Reply:
ah, that's a good idea

Reply:
bah, I gave this a try, the tricky bit is each method pulls the attributes from slightly different places, since, e.g. the delete info doesn't have an actual study object to use.  So refactoring it made the code more complex.

Reply:
Oh well... worth a shot